### PR TITLE
Add SauceLabs Functional Page Loading

### PR DESCRIPTION
**Pull request by alex-nov-git**
Added: 4 independent tests in one class, that provide tests for
admin, order, frontend and customers pages. Each test checks
all pages of its type, and then checks if one or more pages crach.
TODO: remove creating customer from all tests and move it to the Fixture.

### CODE REVIEW

File: tests/Magento_Page_Loading.php [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+        [
+            'browserName' => 'chrome',
+            'desiredCapabilities' => [
+                'version' => '45.0',
+                'platform' => 'OS X 10.10',
+            ],
+        ],
+    ];
+
+    const ADMIN_USERNAME    = 'denis';
```
Comment (d10nZ) [inline tests/Magento_Page_Loading.php]: please replace it with **user** which will be used only for **test** purposes
such functionality should be moved into separate config.php file

**Review by d10nZ — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+	make -j magento_cart magento_registration magento_page_loading
 
 magento_cart:
 	vendor/bin/paratest -p 4 -f --phpunit=vendor/bin/phpunit tests/magento_cart.php
 
 magento_registration:
 	vendor/bin/paratest -p 4 -f --phpunit=vendor/bin/phpunit tests/magento_registration.php
+
+magento_page_loading:
+    vendor/bin/paratest -p 4 -f --verbose --phpunit=vendor/bin/phpunit tests/Magento_Page_Loading.php
```
Comment (ghost) [inline Makefile]: I think that is better use 'tab' instead 'spaces'.

```
Tab Makefile
In order for make to tell the difference between a recipe and things that are not a recipe, it uses TAB characters. So, lines that begin with TAB are assumed to be part of a recipe (so they are shell scripts and passed to the shell for parsing), and lines that do not begin with TAB cannot be part of a recipe (so they cannot be shell scripts: they must be make syntax).
```

File: tests/Magento_Page_Loading.php [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    private static $orderFailList       = [];
+
+    /**
+     * Pages that don't use magento layout
+     *
+     * @var array
+     */
+    private $customLayoutPages = [
+        [
+            'link' => '/doneright/enhanced/dashboard',
```
Comment (jonathonbyrdziak) [inline tests/Magento_Page_Loading.php]: The admin url prefix `doneright` should be moved into a separate config file

File: tests/Magento_Page_Loading.php [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+     * @var array
+     */
+    private $frontendLinks = [
+        0 => '/the-science-research',
+        1 => '/testimonials',
+        2 => '/faq',
+        3 => '/blog',
+        4 => '/ingredients',
+        5 => '/relieffactor-quickstart-pack.html',
+    ];
```
Comment (jonathonbyrdziak) [inline tests/Magento_Page_Loading.php]: separate config file for the list of pages to test

File: tests/Magento_Page_Loading.php [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+    {
+        $this->adminLogin();
+
+        $this->customAdminPagesLoading();
+
+        $links = $this->getAllAdminLinks();
+
+        if (!is_array($links) || empty($links)) {
+            $this->fail('Admin links now found');
+        }
```
Comment (jonathonbyrdziak) [inline tests/Magento_Page_Loading.php]: should be "Admin links *not* found"

**Review by jonathonbyrdziak — CHANGES_REQUESTED** [review]

**Comment by jonathonbyrdziak** [comment]
Looking good so far.

File: tests/Magento_Page_Loading.php (hunk 3 above)
Comment (d10nZ) [reply tests/Magento_Page_Loading.php]: moreover we should move into separate config file all specific data and parameters which can be used by all of us (config.local.php) and by travis CI service (config.staging.php)

domain name => ''
admin domain name => ''
preferred browsers => []
admin_account => [login => 'test', pass=> 'test']
customer_account => [login => 'test', pass=> 'test']
etc

File: tests/Magento_Page_Loading.php (hunk 4 above)
Comment (d10nZ) [reply tests/Magento_Page_Loading.php]: it is better to use dataProvider for such purposes
see: https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html
   chapter: 'Data Providers'

File: tests/Magento_Page_Loading.php [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+                    "Failed frontend pages - %d\n%s",
+                    count(self::$frontFailList),
+                    print_r(self::$frontFailList, true)
+                )
+            );
+        }
+    }
+
+    /**
+     * Testing of pages loading in customer account
```
Comment (alex-nov-git) [inline tests/Magento_Page_Loading.php:140]: Fix comment

**Review by d10nZ — APPROVED** [review]

File: tests/Magento_Page_Loading.php (hunk 3 above)
Comment (alex-nov-git) [reply tests/Magento_Page_Loading.php]: I moved these links into separate configuration file. We can use this file as a global configuration for testing module. But at the same time these links may seem irrelevant in the global configuration file. The data provider method is simple method to provide arguments for test method, so we should understand that the data provider and separate links in config file is a different things. We can use links from separate config file or data provider, that returns array with hard coded links for this test case, or data provider, that returns array with links from configuration, what do you think?

**Review by jonathonbyrdziak — APPROVED** [review]

**Pull request merged**